ulmfit/pretrain_lm.py

```diff
@@ -116,17 +126,11 @@ def pretrain_lm(dir_path, lang='en', cuda_id=0, qrnn=True, clean=True, max_vocab
     learn.opt_fn = partial(optim.Adam, betas=(0.8, 0.99))
     learn.true_wd = False
 
-    # save vocabulary
-    print(f"Saving vocabulary as {dir_path / model_dir}")
-    with open(dir_path / model_dir / f'itos_{name}.pkl', 'wb') as f:
-        pickle.dump(itos, f)
-
     fit_one_cycle(learn, num_epochs, 5e-3, (0.8, 0.7), wd=1e-7)
 
-    if clean and max_vocab is None:
+    if not subword and max_vocab is None:
```

Comment:
you can  calculate perplexity by multiplying cross_entropy by tokens/words ratio and then putting that to exp

Ie something like this:
```
num_tokens = ...;
num_words = ...
tokens_fraction  = num_tokens / num_words

perplexity=np.exp(logloss * tokens_to_words)
```

Here is how to get  tokens_fraction:
https://github.com/n-waves/fastai/blob/poleval/task3/courses/dl2/imdb_scripts/pretrain_lm.py#L51-L55

Here the `logloss` might not be calculated as we would want to but so the result might not be 100% correct  but we can get an estimate like that for the time being. Later we can double check that the math is correct in the perplexity calculation.

Reply:
ok let's  have it implemented at a later time.